File: src/types/patient.ts
/**
 * 患者管理相关类型定义
 * Patient management related type definitions
 */

// 患者基本信息
export interface Patient {
  id: string
  name: string
  age: number
  gender: 'male' | 'female'
  phone: string
  idCard?: string
  avatar?: string
  tags: string[]
  lastVisit?: Date
  medicalHistory: MedicalRecord[]
  createdAt: Date
  updatedAt: Date
}

// 病历记录
export interface MedicalRecord {
  id: string
  patientId: string
  consultationId: string
  diagnosis: string
  symptoms: string[]
  prescription: Prescription[]
  examinations: Examination[]
  doctorId: string
  doctorName: string
  createdAt: Date
}

// 处方信息
export interface Prescription {
  id: string
  medicationName: string
  dosage: string
  frequency: string
  duration: string
  instructions: string
}

// 检查报告
export interface Examination {
  id: string
  type: string
  name: string
  result: string
  fileUrl?: string
  reportDate: Date
}

// 患者查询参数
export interface PatientQuery {
  keyword?: string
  tags?: string[]
  gender?: 'male' | 'female'
  ageRange?: {
    min: number
    max: number
  }
  lastVisitRange?: {
    start: Date
    end: Date
  }
  page: number
  pageSize: number
}

// 患者列表响应
export interface PatientList {
  patients: Patient[]
  total: number
  page: number
  pageSize: number
}

// 患者详情
export interface PatientDetail extends Patient {
  consultationHistory: ConsultationSummary[]
  followUpReminders: FollowUpReminder[]
}

// 问诊摘要
export interface ConsultationSummary {
  id: string
  type: 'text' | 'video' | 'phone'
  status: 'active' | 'completed' | 'cancelled'
  diagnosis?: string
  createdAt: Date
  completedAt?: Date
}

// 随访提醒
export interface FollowUpReminder {
  id: string
  patientId: string
  type: 'checkup' | 'medication' | 'lifestyle'
  message: string
  scheduledAt: Date
  completed: boolean
}

// 患者筛选条件
export interface PatientFilters {
  tags: string[]
  gender?: 'male' | 'female'
  ageRange?: {
    min: number
    max: number
  }
  hasRecentVisit: boolean
}

// 患者服务接口
export interface PatientService {
  getPatientList(query: PatientQuery): Promise<PatientList>
  getPatientDetail(patientId: string): Promise<PatientDetail>
  updatePatientTags(patientId: string, tags: string[]): Promise<void>
  searchPatients(keyword: string): Promise<Patient[]>
  addFollowUpReminder(
    patientId: string,
    reminder: Omit<FollowUpReminder, 'id' | 'patientId'>
  ): Promise<void>
}

// 患者状态管理
export interface PatientState {
  patients: Patient[]
  selectedPatient: Patient | null
  patientDetail: PatientDetail | null
  searchQuery: string
  filters: PatientFilters
  loading: boolean
  error: string | null
  pagination: {
    page: number
    pageSize: number
    total: number
  }
}

File: src-tauri/src/models/patient.rs
// 患者模型

use serde::{Deserialize, Serialize};
use chrono::{DateTime, Utc};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Patient {
    pub id: String,
    pub name: String,
    pub age: Option<u32>,
    pub gender: Option<String>,
    pub phone: Option<String>,
    #[serde(rename = "idCard")]
    pub id_card: Option<String>,
    pub tags: Vec<String>,
    #[serde(rename = "avatarUrl")]
    pub avatar_url: Option<String>,
    #[serde(rename = "lastSync")]
    pub last_sync: Option<DateTime<Utc>>,
    #[serde(rename = "createdAt")]
    pub created_at: DateTime<Utc>,
    #[serde(rename = "updatedAt")]
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum Gender {
    Male,
    Female,
}

impl std::fmt::Display for Gender {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        match self {
            Gender::Male => write!(f, "male"),
            Gender::Female => write!(f, "female"),
        }
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PatientQuery {
    pub keyword: Option<String>,
    pub tags: Option<Vec<String>>,
    pub gender: Option<Gender>,
    #[serde(rename = "ageRange")]
    pub age_range: Option<AgeRange>,
    #[serde(rename = "lastVisitRange")]
    pub last_visit_range: Option<DateRange>,
    pub page: u32,
    #[serde(rename = "pageSize")]
    pub page_size: u32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AgeRange {
    pub min: u32,
    pub max: u32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DateRange {
    pub start: DateTime<Utc>,
    pub end: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PatientList {
    pub patients: Vec<Patient>,
    pub total: u32,
    pub page: u32,
    #[serde(rename = "pageSize")]
    pub page_size: u32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PatientDetail {
    #[serde(flatten)]
    pub patient: Patient,
    #[serde(rename = "consultationHistory")]
    pub consultation_history: Vec<ConsultationSummary>,
    #[serde(rename = "followUpReminders")]
    pub follow_up_reminders: Vec<FollowUpReminder>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConsultationSummary {
    pub id: String,
    #[serde(rename = "type")]
    pub consultation_type: String,
    pub status: String,
    pub diagnosis: Option<String>,
    #[serde(rename = "createdAt")]
    pub created_at: DateTime<Utc>,
    #[serde(rename = "completedAt")]
    pub completed_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FollowUpReminder {
    pub id: String,
    #[serde(rename = "patientId")]
    pub patient_id: String,
    #[serde(rename = "type")]
    pub reminder_type: String,
    pub message: String,
    #[serde(rename = "scheduledAt")]
    pub scheduled_at: DateTime<Utc>,
    pub completed: bool,
}